concordium-std/src/impls.rs

```diff
@@ -942,36 +991,30 @@ impl Read for ExternParameter {
 impl HasParameter for ExternParameter {
     #[inline(always)]
     // parameter 0 always exists so this is correct
-    fn size(&self) -> u32 { unsafe { get_parameter_size(0) as u32 } }
+    fn size(&self) -> u32 {
+        self.get_parameter_size(0).unwrap_abort()
+    }
 }
 
+impl HasFunctionsAPIWrapper for ExternParameter {}
+
 /// The read implementation uses host functions to read chunks of return value
 /// on demand.
 impl Read for CallResponse {
     fn read(&mut self, buf: &mut [u8]) -> ParseResult<usize> {
-        let len: u32 = {
-            match buf.len().try_into() {
-                Ok(v) => v,
-                _ => return Err(ParseError::default()),
-            }
-        };
-        let num_read = unsafe {
-            get_parameter_section(self.i.into(), buf.as_mut_ptr(), len, self.current_position)
-        };
-        if num_read >= 0 {
-            self.current_position += num_read as u32;
-            Ok(num_read as usize)
-        } else {
-            Err(ParseError::default())
-        }
+        let num_read = self.get_parameter_section(self.i.into(), buf, self.current_position).map_err(|_| ParseError::default())?;
+        self.current_position += num_read as u32;
+        Ok(num_read as usize)
     }
 }
 
+impl HasFunctionsAPIWrapper for CallResponse {}
```

Comment:
I don't get these implementations. `CallResponse` should only have Read + Seek implementation. That is the only thing you can do with it.